### PR TITLE
Implement `includeTotal` option for `get`.

### PR DESCRIPTION
**Pull request by brianloveswords**
This came up in a discussion on mozilla/badgekit-api#113 and it seemed like a generically useful thing to provide.

cc @andrewhayward, @christensenep

### CODE REVIEW

**Comment by brianloveswords** [comment]
Can't wait for TravisCI to arbitrarily fail this (tests pass locally)

**Comment by ghost** [comment]
Oh, nice!  Happytimes.

**Comment by brianloveswords** [comment]
One thing to note (which is noted in the updated docs) – when using `includeTotal` an object, instead of an array, will be passed back, e.g.

``` js
{ total: 39201, rows: [ {id: 1}, {id: 2} ] }
```

instead of just

``` js
[ {id: 1}, {id: 2} ]
```

File: index.js [hunk 1]
```diff
@@ -207,9 +208,19 @@ tableProto.get = function get(cnd, opts, callback) {
...
+
+      const hydratedRows = rows.map(function (row) {
         return new RowClass(row)
-      }))
+      })
+
+      if (!opts.includeTotal)
+        return resolver.resolve(hydratedRows)
+
+      self.getOne('SELECT COUNT(*) AS `total` FROM $table', function (err, data) {
```
Comment (andrewhayward) [inline index.js]: Should this not include the same `WHERE` clause used in the original query? Or am I misunderstanding something?

Comment (ghost) [inline index.js]: Yeah, I actually just ran into this.  I think that's what is desired, at least for my nefarious purposes.

Comment (brianloveswords) [inline index.js]: AH INDEED IT SHOULD!

:stars: code reviews :stars:

**Comment by brianloveswords** [comment]
Fixed the `where` issue, let me know if you see anything else stupid.

**Comment by andrewhayward** [comment]
We don't do `GROUP BY`s, so not really an issue, but worth remembering this should that ever get implemented.

**Comment by brianloveswords** [comment]
@andrewhayward Indeed! Does this look ready to go otherwise?

**Comment by andrewhayward** [comment]
@brianloveswords - mentally, yes... haven't had a chance to run through the code outside of my brain.

**Comment by brianloveswords** [comment]
GOOD ENOUGH FOR ME! Tests pass and @christensenep tested it with real world code in mozilla/badgekit-api#113.

**Comment by brianloveswords** [comment]
`v0.8.3` published. (0.8.2 is identical to 0.8.1 because I messed up -_-).

**Pull request merged**